### PR TITLE
[XPU] support noaux_tc

### PR DESCRIPTION
**Pull request by lizan1999**
## Motivation

用于支持EB5，XPU上支持top_method = noaux_tc，新增一个自定义算子fused_noaux_tc。
> :bulb: If this PR is a Cherry Pick, the PR title needs to follow the format by adding the [Cherry-Pick] label at the very beginning and appending the original PR ID at the end. For example, [Cherry-Pick][CI] Add check trigger and logic(#5191)

> :bulb: 如若此PR是Cherry Pick，PR标题需遵循格式，在最开始加上[Cherry-Pick]标签，以及最后面加上原PR ID，例如[Cherry-Pick][CI] Add check trigger and logic(#5191)

## Checklist

- [ ] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [ ] Format your code, run `pre-commit` before commit.
- [ ] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: custom_ops/xpu_ops/src/ops/fused_noaux_tc.cc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+std::vector<paddle::DataType> FusedNoAuxTcInferDtype(
+    const paddle::DataType& gating_logits_dtype,
+    const paddle::DataType& bias_dtype) {
+  return {
+      gating_logits_dtype, paddle::DataType::INT64, paddle::DataType::FLOAT32};
+}
+
+PD_BUILD_STATIC_OP(fused_noaux_tc)
```
Comment (hong19860320) [inline custom_ops/xpu_ops/src/ops/fused_noaux_tc.cc:89]: 是不是需要和GPU算子名称保持一致 ？https://github.com/PaddlePaddle/FastDeploy/blob/16d03c3127f37a9bb8e63b88ebc02c81597f45a4/custom_ops/gpu_ops/noaux_tc.cu#L75

**Review by hong19860320 — APPROVED** [review]
LGTM

**Review by heavengate — APPROVED** [review]

**Review by yongqiangma — APPROVED** [review]
LGTM
